comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala

```diff
@@ -205,12 +205,22 @@ class TcpTransportLayer(port: Int, cert: String, key: String, maxMessageSize: In
     innerBroadcast(peers, msg)
 
   def handleToStream: ToStream => Task[Unit] = {
-    case ToStream(peer, blob) =>
-      withClient(peer, enforce = false) { stub =>
-        stub.stream(Observable.fromIterator(chunkIt(blob)))
-      }.attempt.flatMap {
-        case Left(error) => log.debug(s"Error while streaming packet, error: $error")
-        case Right(_)    => Task.unit
+    case ToStream(peer, path, sender) =>
+      PacketOps.restore[Task](path) >>= {
+        case Right(packet) =>
+          withClient(peer, enforce = false) { stub =>
+            val blob = Blob(sender, packet)
+            stub.stream(Observable.fromIterator(chunkIt(blob)))
+          }.attempt
+            .flatMap {
+              case Left(error) => log.error(s"Error while streaming packet, error: $error")
+              case Right(_)    => Task.unit
+            }
+            .flatMap(kp(Task.delay {
+              if (path.toFile.exists)
+                path.toFile.delete
+            }))
+        case Left(error) => log.error(s"Error while streaming packet, error: $error")
```

Comment:
so a failed restore will never be cleared?

Reply:
you sir are correct, i will issue a ticket to have this functionality tested as well